lib/src/st2110/pipeline/st20_pipeline_tx.c

```diff
@@ -69,6 +74,30 @@ static struct st20p_tx_frame* tx_st20p_next_available(
   return NULL;
 }
 
+static struct st20p_tx_frame* tx_st20p_newest_available(
+    struct st20p_tx_ctx* ctx, enum st20p_tx_frame_status desired) {
+  uint16_t idx = 0;
+  struct st20p_tx_frame* framebuff = NULL;
+  struct st20p_tx_frame* framebuff_newest = NULL;
+
+  for (uint16_t i = 0; i < ctx->framebuff_cnt; i++) {
+    framebuff = &ctx->framebuffs[idx];
+    if ((desired == framebuff->stat &&
+         (!framebuff_newest ||
+          mt_seq32_greater(framebuff->seq_number, framebuff_newest->seq_number)))) {
+      framebuff_newest = framebuff;
+    }
+
+    idx = tx_st20p_next_idx(ctx, idx);
```

Comment:
why do you use `idx` if you anyway iterate through all ctx->framebuff_cnt always from the 0? you can use `i` instead of `idx`.